zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/torrent/BittorrentNotebookRepo.java

```diff
@@ -0,0 +1,257 @@
+    settingsFile = new File(torrentHomeDir.getAbsolutePath() + "settings.dat");
+
+    loadLibrary();
+    setupDir();
+    engine = TorrentEngine.Instance;
+    engine.start();
+    engine.setListener(this);
+  }
+
+  private static void loadLibrary() throws IOException {
+    String path = LIBTORRENT_OS_LIBRARY_PATH();
+    if (path == null) {
+      LOG.error("Unknown Operating system");
+      throw new IOException();
+    }
+    InputStream is = BittorrentNotebookRepo.class.getResourceAsStream(path);
+    File temp = File.createTempFile("libjlibtorrent", ".so");
+    OutputStream os = new FileOutputStream(temp);
+    if (is == null) {
+      throw new FileNotFoundException("File " + path + " was not found inside JAR.");
+    }
+    int readBytes;
+    byte[] buffer = new byte[1024];
+    try {
+      while ((readBytes = is.read(buffer)) != -1) {
+        os.write(buffer, 0, readBytes);
+      }
+    } finally {
+      os.close();
+      is.close();
+    }
+    //System.load(temp.getAbsolutePath());
+    System.setProperty("jlibtorrent.jni.path", temp.getAbsolutePath());
+  }
+
+  private static String LIBTORRENT_OS_LIBRARY_PATH() {
+    String osName = System.getProperty("os.name").toLowerCase();
+    String jvmBits = System.getProperty("sun.arch.data.model");
+    LOG.info("Operating system: " + osName + ", JVM bits: " + jvmBits);
+
+    String ret = null;
+    if (osName.contains("linux")) {
+      if (jvmBits.equals("32")) {
+        ret = "/lib/x86/libjlibtorrent.so";
+      } else {
+        ret = "/lib/x86_64/libjlibtorrent.so";
+      }
+    } else if (osName.contains("windows")) {
+      if (jvmBits.equals("32")) {
+        ret = "/lib/x86/jlibtorrent.dll";
+      } else {
+        ret = "/lib/x86_64/jlibtorrent.dll";
+      }
+    } else if (osName.contains("mac")) {
+      ret = "/lib/x86_64/libjlibtorrent.dylib";
+    }
+    LOG.info("Using libtorrent @ " + ret);
+    return ret;
+  }
+
+  public void addConnection(TorrentSocket torrentSocket) {
+    connections.add(torrentSocket);
+  }
+
+  public void removeConnection(TorrentSocket torrentSocket) {
+    connections.remove(torrentSocket);
+  }
+
+  public void handleMessage(String message) {
+    try {
+      TorrentMessage torrentMessage = TorrentMessage.deserilize(message);
+
+      switch (torrentMessage.op) {
+          case ADDED_TO_DOWNLOAD:
+            break;
+          case DOWNLOAD_COMPLETE:
+            break;
+          case LIST_DOWNLOAD:
+            break;
+          case DOWNLOAD:
+            downloadNote(torrentMessage);
+            break;
+          case METADATA_RECEIVED:
+            break;
+          case STATE_CHANGED:
+            break;
+          case SHARE_NOTE:
+            shareNote(torrentMessage);
+            break;
+      }
+
+    } catch (Exception ex) {
+      LOG.error("Can't handle message" + message);
+    }
+  }
+
+  private void setupDir() {
+    if (!torrentHomeDir.exists())
+      torrentHomeDir.mkdir();
+    if (!torrentDir.exists())
```

Comment:
please make sure you follow the project java style guide regarding [usage of braces]( https://google.github.io/styleguide/javaguide.html#s4.1-braces)